p2panda-rs/src/schema/schema.rs

```diff
@@ -60,6 +61,20 @@ impl Schema {
     pub fn as_cddl(&self) -> String {
         generate_cddl_definition(&self.fields)
     }
+
+    /// Returns a unique string identifier for this schema
+    ///
+    /// This has the format "<schema name>-<hashed schema document view graph tips>".
+    #[allow(unused)]
+    pub fn hash_id(&self) -> String {
+        format!("{}-{}", self.name, self.id.hash().as_str())
```

Comment:
We could introduce something like `SchemaIdHash` here (or a better name..). It would be easy to implement "from_str" or even "Serialize" and "Deserialize" with something like this: https://github.com/p2panda/aquadoggo/blob/4f0deef606c6789b7e22d0c1e86cd3044cf12e7a/aquadoggo/src/graphql/convert/schema.rs#L38

Reply:
The struct could look like that:

```rust
pub struct SchemaIdHash {
    /// Given name of this schema.
    name: String,

    /// Hash of the schema.
    view_hash: DocumentViewHash,
}
```

Reply:
Ah, you mean because then we can use the Rust type system where we want to use it? Good idea.

Reply:
Yeah! We will need it latest in `AbstractQuery`, could be cool to already have the functionality here :+1:

Reply:
I'm not sure why we would make this a struct though. We can't really use the constituent parts for anything except checking equality and storing it as a `String` is much more straightforward to implement.

Reply:
I'm definitely concerned about what we're doing, yeah :laughing:

Reply:
I think it could help to have a definition somewhere WHAT this schema id tries to solve